src/pipeline.cc

```diff
@@ -413,6 +414,43 @@ void Quit(SemaManager &manager) {
   no_active_threads.wait(lock, [] { return !active_threads; });
 }
 
+struct HeadersData {
+  std::size_t content_length;
+};
+
+// return false if end of headers
+void ReadHeaders(bool* eof, HeadersData* headers) {
+  static const std::string_view kContentLength("Content-Length: ");
+  std::string buf;
+  while (true) {
+    int c = getchar();
+    if (c == EOF) {
+      *eof = true;
+      return;
+    }
+    if (c == '\n') {
+      if (buf.empty()) {
+        if (headers->content_length == 0)
```

Comment:
What client did you see that may have such problems?

Reply:
I was writing tool for system tests and ccls was silent